dvc/repo/__init__.py

```diff
@@ -99,7 +99,7 @@ def __init__(self, root_dir=None, scm=None, rev=None):
             )
 
         self.dvc_dir = os.path.join(self.root_dir, self.DVC_DIR)
-        self.config = Config(self.dvc_dir, tree=self.tree)
```

Comment:
I don't know why Config should use a tree with dvcignore.

Reply:
Good point. Looks like a leftover.

Reply:
@karajan1001 ah, wait, it wasn't about dvcignore actually. But about config being able to access config from the git history (GitTree). So we need to rollback this change. Was it causing particular problems?

Reply:
Ok.
Now We had three ignore levels. 
1. Nothing.
2. don't ignore files inside `.dvc` (used for DVC config)
3. ignore all `.dvc` , `.git` , etc, (used by users)

Reply:
@karajan1001 You are right that config doesn't need dvcignore. We just use `self.tree` here as both convenience and necessity. There are really two levels: nothing or `with dvcignore` :slightly_smiling_face: Config clearly doesn't need dvcignore as you've pointed out, but it needs to use repo's tree (of sorts) so it works when we switch to other branches.

Reply:
Seems like it would be nice to have an ability to disable dvcignore on particular tree calls. E.g.
```
with tree.open(..., use_dvcignore=False)
```
or maybe some other convenient way. As I've mentioned, we are in the middle of tree refactorings and we've been discussing whether `exists/isdir/isfile/open` need to care about dvcignore at all (tree.walk clearly does need to care about dvcignore). What do you think?